### PR TITLE
Add scalability goals

### PR DESCRIPTION
**Pull request by jbeda**

### CODE REVIEW

**Comment by jbeda** [comment]
cc @kubernetes/sig-scalability

File: sig-scalability/goals.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+## Introduction
+
+What size clusters do we think that we should support with Kubernetes in the short to medium term?  How performant do we think that the control system should be at scale?  What resource overhead should the Kubernetes control system reasonably consume?  This document attempts to answer these and related questions.
+
+## Background
+Based on our experience running very large production systems at Google, we have found that running excessively large compute and/or storage clusters has several downsides, most notably:
+
+1. **Large-scale cluster failures can lead to large losses of capacity**: Despite our best efforts, correlated cluster-wide hardware or software failures do happen occasionally.  Even with built-in redundancy, load shedding, and good system design practises, the reality is that data center power supplies, and their backup systems do sometimes fail at the same time. Software rollouts, reconfigurations, system overloads and other cascading failures do occasionally go wrong at cluster-wide scale.  When they do (not if they do), it's better not to lose all (or even half of) your total capacity.
```
Comment (timothysc) [inline sig-scalability/goals.md]: I think we should augment some of the comments.  We've already received inquires from folks who would like to push kube to enable some HPC-esk type of workload support.  Typically these environments care less about web-uptime vs. compute utilization of very large single cluster deployments.  

/cc @jeremyeder @ConnorDoyle

File: sig-scalability/goals.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+2. **Scale for the sake of scale can lead to unnecessarily complex system designs**:    Many relatively simple system designs have large, but not infinite, natural scaling limits, and in order to go beyond those limits, more complex designs become necessary.  Unless the need for the additional scale can be justified, the additional complexity often cannot be.
+
+Where Google's internal requirements for compute capacity exceed the natural scaling limits of a cluster (or even multiple clusters, required for redundancy or geo-locality), we typically choose to build more clusters, rather than trying to build larger clusters.  While the resultant increase in the number of clusters can be seen as an unnecessary administrative burden, we have found that the benefits of this approach far outweigh this disadvantage. In addition we have built internal automated tooling which helps us to more easily manage larger numbers of clusters, and accommodate applications that span multiple clusters.
```
Comment (timothysc) [inline sig-scalability/goals.md]: I think this should be more of a best practices / lessons for certain use cases.  Not everyone can afford the many clusters approach, or they may have specialized requirements.  As a community we want to address those use cases where it's feasible.

**Comment by timothysc** [comment]
In general I'd like to shift this from a google-centric doc to a community centric + best practices doc.

**Comment by jbeda** [comment]
@timothysc Can we get this in as is as it is just a transcription from the google doc?  We can then look to review/update it before next meeting?

**Comment by wojtek-t** [comment]
Yeah - I think that Tim`s concerns are valid one. However, I definitely agree that since we already have this doc, we should put it into this repository and then work on improving it.

Thanks Joe for sending this PR.

File: sig-scalability/goals.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+## Open Questions
+
+1. **What, if any, reasonable use cases exist for very large numbers of very small nodes (e.g. for isolation reasons - multitenant)?  Based on comments so far, it seems that the answer is yes, and needs to be addressed.**<br>
+The above scaling goals explicitly accommodate a maximum of 5,000 nodes.  Do we need a special case for larger numbers of small nodes (e.g. 200,000  single-core nodes).  The latter already fits within our other scalability limits (cores per cluster, pods per cluster), so it might not be more difficult to achieve than those.  Two example use cases I've heard anecdotally are (a) running e.g. large numbers of customers' small, largely idle wordpress instances, one per node, and (b) giving away limited numbers of free
```
Comment (porridge) [inline sig-scalability/goals.md]: Truncated sentence: "limited numbers of free".. what? Do you remember @jbeda ?

Comment (wojtek-t) [reply sig-scalability/goals.md]: See the original doc from where it was copied:
https://docs.google.com/document/d/1pABjn69yeDJDl0Yrsneiwj5A_bUI7BZgvKTQtSwzjiE/edit#

**Comment by porridge** [comment]
Thanks, @wojtek-t, https://github.com/kubernetes/community/pull/1201 sent.

File: sig-scalability/goals.md (hunk 3 above)
Comment (quinton-hoole) [reply sig-scalability/goals.md]: Yes, it looks like a cut 'n paste error happened here.  Probably worth fixing.

**Pull request merged**